contracts/evm/contracts/adapters/LayerZeroAdapter.sol

```diff
@@ -0,0 +1,185 @@
+     */
+    function initialize(address _layerZeroEndpoint, address _xCall) public initializer {
+        owner = msg.sender;
+        adminAddress = msg.sender;
+        layerZeroEndpoint = _layerZeroEndpoint;
+        xCall = _xCall;
+    }
+
+    /**
+     * @dev Configure connection settings for a destination chain.
+     * @param networkId The network ID of the destination chain.
+     * @param chainId The chain ID of the destination chain.
+     * @param endpoint The endpoint or address of the destination chain.
+     * @param gasLimit The gas limit for the connection on the destination chain.
+     */
+    function configureConnection(
+        string memory networkId,
+        uint16 chainId,
+        bytes memory endpoint,
+        uint256 gasLimit
+    ) external override onlyAdmin {
+        require(bytes(networkIds[chainId]).length == 0, "Connection already configured");
+        networkIds[chainId] = networkId;
+        chainIds[networkId] = chainId;
+        remoteEndpoint[networkId] = abi.encodePacked(endpoint, address(this));
+        if (gasLimit > 0) {
+            adapterParams[networkId] = abi.encodePacked(uint16(1), gasLimit);
+        } else {
+            adapterParams[networkId] = bytes("");
+        }
+    }
+
+    /**
+     * @dev Get the gas fee required to send a message to a specified destination network.
+     * @param _to The network ID of the target chain.
+     * @param _response Indicates whether the response fee is included (true) or not (false).
+     * @return _fee The fee for sending a message to the given destination network.
+     */
+    function getFee(string memory _to, bool _response) external view override returns (uint256 _fee) {
+        (_fee,) = ILayerZeroEndpoint(layerZeroEndpoint).estimateFees(chainIds[_to], address(this), EMPTY_BYTES, false, adapterParams[_to]);
+    }
+
+    /**
+     * @dev Send a message to a specified destination network.
+     * @param _to The network ID of the destination network.
+     * @param _svc The name of the service.
+     * @param _sn The serial number of the message.
+     * @param _msg The serialized bytes of the service message.
+     */
+    function sendMessage(
+        string memory _to,
+        string memory _svc,
+        int256 _sn,
+        bytes memory _msg
+    ) external override payable {
+        require(msg.sender == xCall, "Only xCall can send messages");
+        uint256 fee;
+
+        if (_sn < 0) {
+            fee = this.getFee(_to, false);
+            if (address(this).balance < fee) {
+                uint256 sn=uint256(- _sn);
+                pendingResponses[sn] = Types.PendingResponse(_msg, _to);
+                emit ResponseOnHold(sn);
+                return;
+            }
+        } else {
+            fee = msg.value;
+        }
+
+
+        ILayerZeroEndpoint(layerZeroEndpoint).send{value: fee}(
+            chainIds[_to],
+            remoteEndpoint[_to],
+            abi.encodePacked(_msg),
+            payable(tx.origin),
```

Comment:
I think if it's a response (when _sn < 0), we can use the contract itself (layerzero adapter contract), and for a new message request, we can send it back to tx.origin.

Reply:
Mabye its easier to just go with the contract always to pad for automatic responses so that we hide the fee, so if we use the airdrop as we discussed it needs to be less exact.